RUFAS/input_manager.py

```diff
@@ -436,117 +441,154 @@ def _validate_json_element(self, element_hierarchy: List[str], properties_blob_k
         """Validates an input data number element."""
         info_map = {"class": self.__class__.__name__,
                     "function": self._num_type_validator.__name__,
                     }
         minimum_value = variable_properties.get("minimum")
         maximum_value = variable_properties.get("maximum")
+        properties_violation_message = f"Violates properties defined in '{properties_blob_key}'."
         if type(input_data_value) is not float and type(input_data_value) is not int:
             warning_string = "Validation: value is not a number."
-            om.add_warning(warning_string, f"{var_name=}", info_map)
+            warning_message = f"Variable '{var_name}' has value: {input_data_value}, is type: " \
+                              f"{type(input_data_value)}. {properties_violation_message}"
+            om.add_warning(warning_string, warning_message, info_map)
             return False
         if minimum_value is not None:
             is_in_range = minimum_value <= input_data_value
             if not is_in_range:
-                warning_string = f"Validation: value less than {minimum_value: .2f}."
-                om.add_warning(warning_string, f"{var_name=}", info_map)
+                warning_name = "Validation: value less than minimum."
+                warning_message = f"Variable '{var_name}' has value: {input_data_value}, less than minimum value: " \
+                                  f"{minimum_value: .2f}. {properties_violation_message}"
+                om.add_warning(warning_name, warning_message, info_map)
                 return False
         if maximum_value is not None:
             is_in_range = input_data_value <= maximum_value
             if not is_in_range:
-                warning_string = f"Validation: value greater than {maximum_value: .2f}."
-                om.add_warning(warning_string, f"{var_name=}", info_map)
+                warning_name = "Validation: value greater than maximum."
+                warning_string = f"Variable '{var_name}' has value: {input_data_value}, greater than maximum value: " \
+                                 f"{maximum_value: .2f}. {properties_violation_message}"
+                om.add_warning(warning_name, warning_string, info_map)
                 return False
 
         return True
 
-    def _string_type_validator(self, variable_properties: Dict[str, Any], var_name: str, input_data_value: str) -> bool:
+    def _string_type_validator(self, variable_properties: Dict[str, Any], var_name: str, input_data_value: str,
+                               properties_blob_key: str) -> bool:
         """Validates an input data string element."""
         info_map = {"class": self.__class__.__name__,
                     "function": self._string_type_validator.__name__,
                     }
+        properties_violation_message = f"Violates properties defined in '{properties_blob_key}'."
```

Comment:
Is the idea that people would know where to find the metadata and would go to look in the metadata and then correct their input? If so, think I'd be more specific that the properties section they're looking for is in the metadata:

```suggestion
        properties_violation_message = f"Violates properties defined in metadata '{properties_blob_key}'."
```

Reply:
Yes, that is generally the idea. Ideally, the messages would specify the top level blob of data in `files` that contains that data in violation of the properties. But making that happen would require some pretty significant changes so I left them out of this PR.